R/packrat.R

```diff
@@ -269,59 +270,77 @@ initImpl <- function(project = getwd(),
+#' to populate the library on that machine.
+#' \item \strong{Sync}: Use \code{restore} to apply library changes made by a
 #' collaborator to your own library. (In general, you want to run \code{restore}
 #' whenever you pick up a change to \code{packrat.lock})
-#'
-#'   \item \strong{Rollback}: Use \code{restore} to undo accidental changes made
+#' \item \strong{Rollback}: Use \code{restore} to undo accidental changes made
 #' to the library since the last snapshot.
 #' }
 #'
 #' \code{restore} cannot make changes to packages that are currently loaded. If
 #' changes are necessary to currently loaded packages, you will need to restart
 #' \R to apply the changes (\code{restore} will let you know when this is
 #' necessary). It is recommended that you do this as soon as possible, because
-#' any library changes made between running \code{restore} and restarting \R will
-#' be lost.
-#'
-#' @note
-#' \code{restore} can be destructive; it will remove packages that were not in
-#' the snapshot, and it will replace newer packages with older versions if
-#' that's what the snapshot indicates. \code{restore} will warn you before
-#' attempting to remove or downgrade a package (if \code{prompt} is
-#' \code{TRUE}), but will always perform upgrades and new installations without
-#' prompting.
-#'
-#' \code{restore} works only on the private package library created by packrat;
-#' if you have other libraries on your path, they will be unaffected.
-#'
-#' The \code{restart} parameter will only result in a restart of R when the
-#' R environment packrat is running within makes available a restart function
-#' via \code{getOption("restart")}.
-#'
-#' @param project The project directory. When in packrat mode, if this is \code{NULL},
-#' then the directory associated with the current packrat project is used. Otherwise,
-#' the project directory specified is used.
+#' any library changes made between running \code{restore} and restarting \R
+#' will be lost.
+#'
+#' @note \code{restore} can be destructive; it will remove packages that were
+#'   not in the snapshot, and it will replace newer packages with older versions
+#'   if that's what the snapshot indicates. \code{restore} will warn you before
+#'   attempting to remove or downgrade a package (if \code{prompt} is
+#'   \code{TRUE}), but will always perform upgrades and new installations
+#'   without prompting.
+#'
+#'   \code{restore} works only on the private package library created by
+#'   packrat; if you have other libraries on your path, they will be unaffected.
+#'
+#'   The \code{restart} parameter will only result in a restart of R when the R
+#'   environment packrat is running within makes available a restart function
+#'   via \code{getOption("restart")}.
+#'
+#'   To install packages hosted in private repositories on GitHub, GitLab, and
+#'   Bitbucket, you must either set the option
+#'   \code{packrat.authenticated.downloads.use.renv} to \code{TRUE} and ensure
```

Comment:
This change isn't included in the NEWS; was that because we were keeping it as an internal option until it got some mileage? I would be happy to see it flip to enabled by default and leave the option in case folks want to revert to the `httr` behavior.

For comparison, we do not explain that `packrat.dependency.discovery.renv` can be used to revert to the old dependency scanning workflow; it remains (temporarily) in case folks find some issue with the new code.

Reply:
As far as I recall, in our discussions, we had considered enabling it by default, but decided to have it default to `FALSE` to change as little as possible.

This preference is already mentioned in the unreleased news.

Reply:
If this option defaulted to `TRUE`, `renv` code would only be used to download when Git provider authentication environment variables are found and when `curl` is available. That will include _all_ downloads from supported Git providers, not just ones in private repos.

I would be fine enabling it, but I feel like we already made this decision, and I can't recall the reasoning.

Reply:
This conversation was resolved in our Sync meeting today. We're going to enable this feature in a separate PR in the future.